src/Perl6/Metamodel/CurriedRoleHOW.nqp

```diff
@@ -79,77 +79,94 @@ class Perl6::Metamodel::CurriedRoleHOW
         nqp::settypecheckmode($type, 2);
     }
 
-    method parameterize_roles($obj) {
-        my @pos_args;
-        nqp::push(@pos_args, $obj);
-        for @!pos_args {
-            nqp::push(@pos_args, $_);
+    # A curried role may be bound to form a role other than its origin, e.g.  a
+    # parametric role group and the candidate we're not quite ready to fetch at
+    # construction time. If we're still not ready yet whenever we need it, the
+    # binding is ourself for now. If the origin lacks a means of producing a
+    # binding, we assume it's its own by default.
+    method bind($obj) {
+        if $!binding =:= NQPMu {
+            if nqp::can($!curried_role.HOW, 'bind') {
+                my $binding := $!curried_role.HOW.bind($!curried_role, $obj);
+                $!binding := $binding unless nqp::decont($binding) =:= $obj;
```

Comment:
If you ever expect `bind` to return a container (which is, perhaps, expected for Raku-land implementations of the method), then deconting must take place the line above or you end up with a container in `$!binding`.